router/src/http/types.rs

```diff
@@ -250,11 +250,36 @@ pub(crate) struct Rank {
 #[derive(Serialize, ToSchema)]
 pub(crate) struct RerankResponse(pub Vec<Rank>);
 
+#[derive(Deserialize, ToSchema, Debug)]
+#[serde(untagged)]
+pub(crate) enum InputType {
+    SingleString(String),
+    SingleInt(u32),
+    VectorInt(Vec<u32>),
+}
+impl InputType {
+    pub(crate) fn count_chars(&self) -> usize {
+        match self {
+            InputType::SingleString(s) => s.chars().count(),
+            InputType::SingleInt(_) => 1,
+            InputType::VectorInt(v) => v.len(),
```

Comment:
Is this how OpenAI counts when ids are given to the API? Or do they still count the chars by decoding the ids?

Reply:
I'll look into it and modify this per my findings.

Reply:
@OlivierDehaene I looked through the source of `openai-python` v1.7.1 and couldn't find a reference to  characters counting in the embeddings API.
Should `count_chars` return `0` for `InputType::SingleInt, InputType::VectorInt` for correctness?
LMK how you want to proceed.